clients/java/src/main/java/marquez/client/MarquezPathV1.java

```diff
@@ -1,76 +1,113 @@
 package marquez.client;
 
 import com.google.common.annotations.VisibleForTesting;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import javax.annotation.Nullable;
 import lombok.NonNull;
 import marquez.client.models.RunState;
 
 class MarquezPathV1 {
-
   @VisibleForTesting static final String BASE_PATH = "/api/v1";
 
   @VisibleForTesting
-  static String path(String pathTemplate, @Nullable String... pathArgs) {
-    return BASE_PATH + String.format(pathTemplate, (Object[]) pathArgs);
+  static List<String> path(String pathTemplate, @Nullable String... pathArgs) {
+    /*
```

Comment:
As I got MarquezPathV1 is for preparing path :)
I would prefer to put encoding here for pathArgs. Cause if it's not encoded path is not ready, then why do we need a class with separate responsibility, which do only half of the task